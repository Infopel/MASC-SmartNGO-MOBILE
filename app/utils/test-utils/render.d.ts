export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
