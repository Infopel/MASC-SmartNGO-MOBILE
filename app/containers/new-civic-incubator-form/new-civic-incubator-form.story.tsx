import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { NewCivicIncubatorForm } from "./new-civic-incubator-form.component"

storiesOf("NewCivicIncubatorForm", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <NewCivicIncubatorForm style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
