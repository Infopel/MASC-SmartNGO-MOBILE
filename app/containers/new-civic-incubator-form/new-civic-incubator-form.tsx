import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { Button } from "components/button/button"
import { FormControlDate } from "components/form-control-date/form-control-date"
import { FormControlInput } from "components/form-control-input/form-control-input"
import { FormControlSeletor } from "components/form-control-seletor/form-control-seletor"
import { Formik, FormikHelpers } from "formik"
import { goBack } from "navigators/navigation-utilities"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { compose } from 'recompose'
import { createInitiative, CreateInitiativeParams } from "storage/mutations/initiative.mutations"
import { findAllDistricts, findAllProvinces, ILocation } from "storage/queries/location"
import * as yup from 'yup'

export interface NewCivicIncubatorFormProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  provinces: ILocation[]
  districts: ILocation[]
}

/**
 * Describe your component here
 */
const Container = function (props: NewCivicIncubatorFormProps) {
  const { style, provinces, districts } = props
  const validationSchema = yup.object().shape({
    adminPost: yup.string().required(),
    dateConstitution: yup.date().required(),
    districtId: yup.string().required(),
    latLng: yup.object().shape({
      latitude: yup.number().required(), longitude: yup.number().required()
    }).required(),
    locality: yup.string().required(),
    name: yup.string().required(),
    provinceId: yup.string().required()
  })



  async function create(values: CreateInitiativeParams, helpers: FormikHelpers<CreateInitiativeParams>) {
    await createInitiative({ ...values, type: 'civic-incubator' })
    goBack()
  }

  return (
    <Formik {...{ initialValues: {}, validationSchema, onSubmit: create }}>
      {({ handleSubmit }) => <>
        <FormControlInput label='new-civic-incubator.productName' fieldName="name" />
        <FormControlSeletor fieldName="provinceId" testID="province" label='new-civic-incubator.province' placeholder={'common.select-an-item'} items={provinces} />
        <FormControlSeletor fieldName="districtId" testID="district" label='new-civic-incubator.district' placeholder={'common.select-an-item'} items={districts} />
        <FormControlInput label='new-civic-incubator.admin-post' fieldName="adminPost" />
        <FormControlInput label='new-civic-incubator.locality' fieldName="locality" />
        <FormControlInput label='new-civic-incubator.longitude' fieldName='latLng.longitude' />
        <FormControlInput label='new-civic-incubator.latitude' fieldName='latLng.latitude' />
        <FormControlDate label='new-civic-incubator.date-constitution' fieldName='dateConstitution' />
        <Button tx="common.save" testID="save" onPress={handleSubmit.bind(this, undefined)}></Button>
      </>}
    </Formik>
  )
}

export const NewCivicIncubatorFormContainer = compose(
  withDatabase,
  withObservables(['database'], ({ database }) => ({
    provinces: findAllProvinces(database), districts: findAllDistricts(database)
  }))
)(Container)