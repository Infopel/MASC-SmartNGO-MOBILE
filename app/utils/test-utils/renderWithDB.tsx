jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  