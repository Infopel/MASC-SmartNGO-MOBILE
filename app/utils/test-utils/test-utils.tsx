import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

