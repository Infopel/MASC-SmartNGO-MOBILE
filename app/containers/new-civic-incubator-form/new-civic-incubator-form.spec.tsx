import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { NewCivicIncubatorFormContainer } from "./new-civic-incubator-form"

describe("<NewCivicIncubatorFormContainer>",()=>{
 it("render", async function () {
    const component = <NewCivicIncubatorFormContainer></NewCivicIncubatorFormContainer>
    
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
  