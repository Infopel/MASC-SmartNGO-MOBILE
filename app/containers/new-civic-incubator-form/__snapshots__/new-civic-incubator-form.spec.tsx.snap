// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewCivicIncubatorFormContainer> render 1`] = `null`;
